packages/database/src/core/util/libs/parser.ts

```diff
@@ -119,13 +138,21 @@ export const parseURL = function(
       dataURL = dataURL.substring(colonInd + 2);
     }
 
-    // Parse host and path.
+    // Parse host, path, and query string.
     let slashInd = dataURL.indexOf('/');
     if (slashInd === -1) {
       slashInd = dataURL.length;
     }
-    host = dataURL.substring(0, slashInd);
-    pathString = decodePath(dataURL.substring(slashInd));
+    let questionInd = dataURL.indexOf('?');
```

Comment:
Nit: Can you make this `questionMarkInd`?

Reply:
done